### ISSUE TITLE
This is fantastic!! Where do you need help?

### ISSUE THREAD

**Issue by LarsKemmann** [jjmccollum/open-cbgm#2]
I cannot begin to tell you how excited I am to have found an OSS implementation of CBGM, and on top of that one that is/promises to be compatible with the data & local/global decisions collected by INTF and others!

You mention a couple of items in your roadmap for future development. Where do you need help?

E.g., would it be helpful to design and develop a web app front-end that provides the non-technical user with a UI for interacting with CBGM and various data collections, and perhaps an API that provides the more technical user with a means to integrate data collections and extract results via HTTP(S) requests? That's the area I'm most focused on, but I have enough background in C++ to be able to bridge the two as needed. I would also love to see an open dataset (with some means of tracing through and authenticating/verifying the contributions to that dataset) so that there can be open collaboration and interaction with CBGM.

Anyways, I think it would be great if you could create some issues in the repository here and tag them with "contributions-welcome" or something like that. I'd love to see if there is a way for me (and others) to get involved!

**Comment by jjmccollum** [comment]
@LarsKemmann, I'm delighted to hear that you're interested in the open-cbgm library and curious about how to help with future development!

The two items listed in the roadmap (a convenient utility for taking the TEI XML output of a collation tool and formatting it for analysis with the CBGM, and a more user-friendly interface for the library's modules) are still on the table, and I'd greatly appreciate help on either of them. In both cases, a GUI that would facilitate usage for non-technical users would be ideal. I originally intended to outline some of the broader implementation ideas I had for these items below, but my outlines ended up getting too detailed, so I'll provide them in dedicated issues, per your suggestion. To give a short answer, it would absolutely be helpful to have a graphical front end interface like the one you've described!

I'll admit, I'm relatively new to the GitHub scene, so I'm not exactly sure what the best practice is for collaborative efforts like this.  I'd prefer to keep this repository limited to the core library, so I think it would be best if we created separate repositories for external interfaces. You'll probably be doing most of the heavy lifting (since GUIs are not my strong suit), so you would be welcome to author such a repository. I'd be happy to contribute, to the extent that I would be any help, but the credit should definitely go to you. (If you're more familiar with this aspect of GitHub, feel free to correct any misconceptions I have!)

Any continued discussion is welcome here, but implementation details of specific features should be discussed in the issues for those features.

**Comment by LarsKemmann** [comment]
Coming back to this after taking some time to think through the issues involved... First of all thank you for adding the feature issues for discussion, that's great!

As I'm learning more about CBGM, I'm running into the challenge (which probably comes as no surprise to you) that so much of the terminology, ecosystem, etc. is new. Like what TEI is. Do you know of a good introductory resource for that, or ideally a tutorial for using different things together? Like a few worked examples from [A New Approach to Textual Criticism](https://www.amazon.com/New-Approach-Textual-Criticism-Coherence-Based/dp/0884142671/ref=tmm_hrd_swatch_0?_encoding=UTF8&qid=&sr=)? I think once I actually get to work through a tutorial I'd be able to understand how to put the pieces together in a mental model and then I could understand how best to think through the new features/user interfaces.

**Comment by jjmccollum** [comment]
Thanks! The best reference I know of for TEI is the official guidelines (https://www.tei-c.org/release/doc/tei-p5-doc/en/html/index.html), but for examples, I've found that the outputs of the ITSEE Transcription Editor (available online at https://itsee-wce.birmingham.ac.uk/ote/transcriptiontool or for local download at https://sourceforge.net/projects/wfce-ote/), especially if you compare them to what you see in the interface, are instructive. The TEI XML collation files in the `examples` folder of this repository are similar, but they are stripped down to the minimal TEI XML elements needed for processing in the CBGM.

In the interest of avoiding duplicated effort, I should let you know now that two very talented developers reached out to me a while ago to express their interest in collaborating on a web interface for this library, so I've already started working on this component with them. It may be a while before we have something complete up and running, as we're doing this with our spare time, but we're making steady progress!
